wallet/account.go

```diff
@@ -0,0 +1,71 @@
+package wallet
+
+import (
+	"crypto/ed25519"
+	"fmt"
+
+	"github.com/iotaledger/hive.go/crypto"
+	iotago "github.com/iotaledger/iota.go/v4"
+)
+
+// Account represents an account.
+type Account interface {
+	// ID returns the accountID.
+	ID() iotago.AccountID
+
+	// Address returns the account address.
+	Address() iotago.Address
+
+	// PrivateKey returns the account private key for signing.
+	PrivateKey() ed25519.PrivateKey
+}
+
+var _ Account = &Ed25519Account{}
+
+// Ed25519Account is an account that uses an Ed25519 key pair.
+type Ed25519Account struct {
+	accountID  iotago.AccountID
+	privateKey ed25519.PrivateKey
+}
+
+// NewEd25519Account creates a new Ed25519Account.
+func NewEd25519Account(accountID iotago.AccountID, privateKey ed25519.PrivateKey) *Ed25519Account {
+	return &Ed25519Account{
+		accountID:  accountID,
+		privateKey: privateKey,
+	}
+}
+
+// ID returns the accountID.
+func (e *Ed25519Account) ID() iotago.AccountID {
+	return e.accountID
+}
+
+// Address returns the account address.
+func (e *Ed25519Account) Address() iotago.Address {
```

Comment:
That confused me a bit.
It's not the account address derived from the accountID, but the Ed25519 address derived from the private key. :sweat_smile: 
Can we change the function name and/or the comment?

Reply:
That is why I told you to wait with the review, exactly because I was refactoring this, hehe